packages/clay-color-picker/src/Custom.tsx

```diff
@@ -177,19 +177,23 @@ const ClayColorPickerCustom: React.FunctionComponent<IProps> = ({
 		onChange(hexString);
 
 		if (setInput) {
-			setHexInput(colorValue.toHex());
+			setHexInput(hexString);
 		}
 	};
 
 	React.useEffect(() => {
 		if (inputRef.current !== document.activeElement) {
-			setHexInput(color.toHex());
-
-			if (!showPalette) {
+			if (color.isValid()) {
 				setHue(color.toHsv().h);
+
+				if (!colors.includes(color.toHex())) {
+					setNewColor(color);
+				} else {
+					setHexInput(color.toHex());
+				}
```

Comment:
It seems that this is having a side effect when interacting with the Hue component and the Gradient selector when moving to a black color and in some situations the Hue is also changing color when moving through the Gradient.

See this video:

https://user-images.githubusercontent.com/13750819/155434784-ceb13a73-785e-4a16-a078-e1457e990f1f.mp4